scripts/locust/service.py

```diff
@@ -40,120 +81,251 @@ def parallel_requests():
     group.join()
 
 
-class ServiceTaskSet(TaskSet):
-    """This is the default page so start here"""
+# These task-less parent classes define common methods involving the locust HTTP
+# client.
+class AzulTaskSet(SequentialTaskSet):
+    catalog: str = configured_catalog()
 
-    def on_start(self):
-        self.projects_page()
+    def endpoint(self, path: str, **query) -> str:
+        return furl(path=path, query=dict(query, catalog=self.catalog)).url
 
-    def projects_page(self):
-        with parallel_requests() as group:
-            group.spawn(lambda: self.client.get('/index/summary?filters=%7B%7D'))
-            group.spawn(lambda: self.client.get('/index/projects?filters=%7B%7D&size=15'))
 
-    @task
-    def filter_mtx_files(self):
+class BrowserTaskSet(AzulTaskSet):
+    browser_search_params = dict(size=15, sort='entryId', order='desc')
```

Comment:
What determines if this is used or not? I find it very hard to follow the logic.

Reply:
It's determined by whether the tasks use any browser-specific methods or settings. The task sets for the repository and DRS endpoints don't use them. Perhaps they should? It's unclear to me when/how these endpoints are invoked by the browser.